### DISCUSSION TITLE
Errors from new version of MOOSE&SAM

### DISCUSSION THREAD

**Discussion by Terryxixi2014** [idaholab/moose#19284] [Q&A General] [answered]
Any comments on the following errors and warnings?

*** Warning, This code is deprecated and will be removed in future versions:
The parameter accept_on_max_picard_iteration is deprecated.
True to treat reaching the maximum number of Picard iterations as converged.
Stack frames: 14
0: libMesh::print_trace(std::ostream&)
1: void moose::internal::mooseDeprecatedStream<libMesh::BasicOStreamProxy<char, std::char_traits<char> >, char const (&) [15], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, char const (&) [17], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&>(libMesh::BasicOStreamProxy<char, std::char_traits<char> >&, bool, char const (&) [15], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, char const (&) [17], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)
2: Parser::extractParams(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, InputParameters&)
3: Parser::walkRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, hit::Node*)
4: Parser::walk(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, hit::Node*)
5: hit::Node::walk(hit::Walker*, hit::NodeType)
6: hit::Node::walk(hit::Walker*, hit::NodeType)
7: Parser::parse(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)
8: MooseApp::setupOptions()
9: SamApp::setupOptions()
10: MooseApp::run()
11: ./sam-opt(+0x5153) [0x55afe87de153]
12: __libc_start_main
13: ./sam-opt(+0x53dd) [0x55afe87de3dd]

Create mesh... OK
Build SAM Displaced Mesh... OK
Checking system integrity... OK
Add Variables... OK
Initialize Variable Values... OK

*** ERROR ***
basic_string::substr: __pos (which is 18446744073709551615) > this->size() (which is 20)

Stack frames: 7
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: void mooseError<char const*>(char const*&&)
3: /home/jwang/Projects/SAM00/moose/framework/libmoose-opt.so.0(+0x72a16d) [0x7f264bfc416d]
4: ./sam-opt(+0x5153) [0x55afe87de153]
5: __libc_start_main
6: ./sam-opt(+0x53dd) [0x55afe87de3dd]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor

**Comment by GiudGiud** [comment, marked as answer]
Hello

The warning is because the parameter is now called accept_on_max_fixed_point_iteration. 

The error is some out of bounds error. Please compile SAM in debug mode and get a full backtrace for the error. I have access to SAM I can have a look. If it s not obvious we ll tag some SAM developers in here.

Guillaume

**Reply by GiudGiud** [reply]
please avoid posting source code btw. Just email me at guillaume.giudicelli@inl.gov
